src/main/kotlin/com/sourcegraph/cody/config/ui/CodyConfigurable.kt

```diff
@@ -133,9 +143,37 @@ class CodyConfigurable(val project: Project) : BoundConfigurable(ConfigUtil.CODY
     codyApplicationSettings.blacklistedLanguageIds = settingsModel.blacklistedLanguageIds
     codyApplicationSettings.shouldAcceptNonTrustedCertificatesAutomatically =
         settingsModel.shouldAcceptNonTrustedCertificatesAutomatically
+    applyChannelConfiguration()
 
     publisher.afterAction(context)
   }
+
+  private fun applyChannelConfiguration() {
+    val configuredChannel = findConfiguredChannel()
+    val newChannel = settingsModel.channel
+
+    if (!configuredChannel.equals(newChannel)) {
+      if (!UpdateChannel.Stable.equals(configuredChannel)) {
+        UpdateSettings.getInstance().storedPluginHosts.remove(configuredChannel.channelUrl)
+      }
+
+      if (!UpdateChannel.Stable.equals(newChannel)) {
+        UpdateSettings.getInstance().storedPluginHosts.add(newChannel.channelUrl)
```

Comment:
Is there a way to trigger an update when you switch so that users get some visual hint to update? Alternatively, is it possible to link to the "Plugins" section after the user clicks on `Apply` so that users can figure out what to do. Worst case, we should have some paragraph explaining that the user needs to restart to pick up an update, and then update, and then restart again.